src/app/shared/chart-and-table/chart-and-table.component.ts

```diff
@@ -0,0 +1,189 @@
+import { Component, OnInit, Input } from '@angular/core';
+import * as Highcharts from 'highcharts';
+import _sortBy from 'lodash-es/sortBy';
+import * as _ from 'lodash';
+// import chroma from 'chroma-js';
+// import * as moment from 'moment';
+// import { GRAPH_COLUMN_NAME } from '../constants/graphConstants';
+// import { DAY_DATE_FORMAT, DATE_FORMAT } from '../constants/siteConstants';
+// import { formatNumber } from '../common-utils/common-utils';
+
+@Component({
+  selector: 'app-chart-and-table',
+  templateUrl: './chart-and-table.component.html',
+  styleUrls: ['./chart-and-table.component.scss']
+})
+export class ChartAndTableComponent implements OnInit {
+
+  @Input() setChartTheme: any = [];
+  @Input() chartType: string = 'spline';
+  @Input() chartTitle: string = null;
+  @Input() chartSubTitle: string = null;
+  @Input() xAxisCategories: Array<string> = [];
+  @Input() chartSeries: Array<object> = [];
+  @Input() columnDefs: Array<object> = [];
+  @Input() tableData: Array<object> = [];
+
+  public Highcharts: typeof Highcharts = Highcharts; // Highcharts, it's Highcharts
+  public chartOptions: object = {}
+  private yAxisConfig: any = [];
+
+  // TODO:
+  private graphMeta = {};
+  private selectorGroups = [];
+  private selectors = [];
+  private config = {};
+
+  private GRAPH_SERIE_OPTIONS_DEFAULT = {
+    connectNulls: true
+  };
+
+  private GRAPH_XAXIS_OPTIONS_DEFAULT = {
+    title: {
+      text: null
+    }
+  };
+
+  private GRAPH_YAXIS_OPTIONS_DEFAULT = {
+    title: {
+      text: null
+    }
+  };
+
+  constructor() { }
+
+  ngOnInit(): void {
+  
+    this.chartInit();
+  }
+
+  private initialize() {
+
+  }
+
+  private chartInit() {
+    this.Highcharts.setOptions(this.setChartTheme);
+    this.setYAxis();
+    this.chartOptions = {
```

Comment:
RFC: What do you think about moving these styles into separate file? In our existing project, we have it exactly as you have it there. I think it distract from reading the "real" code but it will require to pass arguments. So I'm not sure if it is worth it (or we can just postpone it).

Reply:
Yes, I think this is good to place the graph style in a separate file because with this approach we set style only in one place and the data related config in another place. For this, it's easy to maintain a graph in the future.